File: core/src/com/broll/gainea/server/core/actions/RequiredActionContext.java
package com.broll.gainea.server.core.actions;

import com.broll.gainea.net.NT_Action;
import com.broll.gainea.net.NT_PlayerAction;

public class RequiredActionContext<T extends NT_Action> extends ActionContext<T> {

    private ActionContext<T> actionContext;
    private String text;
    private Object messageForOtherPlayer;

    public RequiredActionContext(ActionContext<T> actionContext, String text) {
        this(actionContext, text, null);
    }

    public RequiredActionContext(ActionContext<T> actionContext, String text, Object messageForOtherPlayers) {
        super(actionContext.getAction());
        this.actionContext = actionContext;
        this.text = text;
        this.messageForOtherPlayer = messageForOtherPlayers;
    }

    public Object getMessageForOtherPlayer() {
        return messageForOtherPlayer;
    }

    public NT_PlayerAction nt() {
        NT_PlayerAction action = new NT_PlayerAction();
        action.action = actionContext.getAction();
        action.text = text;
        return action;
    }

    public ActionContext<T> getActionContext() {
        return actionContext;
    }
}

File: server/src/main/com/broll/gainea/server/core/actions/RequiredActionContext.kt
package com.broll.gainea.server.core.actions

import com.broll.gainea.net.NT_Action
import com.broll.gainea.net.NT_PlayerAction

class RequiredActionContext<T : NT_Action?> @JvmOverloads constructor(val actionContext: ActionContext<T>, val text: String?, val messageForOtherPlayer: Any? = null) : ActionContext<T>(actionContext.action) {

    fun nt(): NT_PlayerAction {
        val action = NT_PlayerAction()
        action.action = actionContext.action
        action.text = text
        return action
    }
}
